python/graphstorm/run/gsgnn_emb/gsgnn_node_emb.py

```diff
@@ -0,0 +1,113 @@
+                                      train_etypes=config.train_etype,
+                                      eval_etypes=config.eval_etype,
+                                      node_feat_field=config.node_feat_name,
+                                      pos_graph_feat_field=config.lp_edge_weight_for_loss)
+    elif config.task_type in {BUILTIN_TASK_NODE_REGRESSION, BUILTIN_TASK_NODE_CLASSIFICATION}:
+        train_data = GSgnnNodeTrainData(config.graph_name,
+                                        config.part_config,
+                                        train_ntypes=config.target_ntype,
+                                        eval_ntypes=config.eval_target_ntype,
+                                        node_feat_field=config.node_feat_name,
+                                        label_field=config.label_field)
+    elif config.task_type in {BUILTIN_TASK_EDGE_CLASSIFICATION, BUILTIN_TASK_EDGE_REGRESSION}:
+        train_data = GSgnnEdgeTrainData(config.graph_name,
+                                        config.part_config,
+                                        train_etypes=config.target_etype,
+                                        node_feat_field=config.node_feat_name,
+                                        label_field=config.label_field,
+                                        decoder_edge_feat=config.decoder_edge_feat)
+    else:
+        raise TypeError("Not supported for task type: ", config.task_type)
+
+    # assert the setting for the graphstorm embedding generation.
+    assert config.save_embed_path is not None, \
+        "save embeded path cannot be none for gs_gen_node_embeddings"
+    assert config.restore_model_path is not None, \
+        "restore model path cannot be none for gs_gen_node_embeddings"
+
+    if config.task_type == BUILTIN_TASK_LINK_PREDICTION:
+        model = gs.create_builtin_lp_gnn_model(train_data.g, config, train_task=False)
+    elif config.task_type in {BUILTIN_TASK_NODE_REGRESSION, BUILTIN_TASK_NODE_CLASSIFICATION}:
+        model = gs.create_builtin_node_gnn_model(train_data.g, config, train_task=False)
+    elif config.task_type in {BUILTIN_TASK_EDGE_CLASSIFICATION, BUILTIN_TASK_EDGE_REGRESSION}:
+        model = gs.create_builtin_edge_gnn_model(train_data.g, config, train_task=False)
+
+    model_path = config.restore_model_path
+    # TODO(zhengda) the model path has to be in a shared filesystem.
+    model.restore_model(model_path)
+    # Preparing input layer for training or inference.
+    # The input layer can pre-compute node features in the preparing step if needed.
+    # For example pre-compute all BERT embeddings
+    model.prepare_input_encoder(train_data)
+    # Runjie: To generate embeddings, it might be preferable to utilize the
+    # entire graph instead of just the training graph.
+    # Additionally, generating embeddings will not result in any edge leakage issues.
+    embeddings = do_full_graph_inference(model, train_data, fanout=config.eval_fanout,
```

Comment:
In some cases, we can only do mini-batch inference.
Please check the implementation of different tasks.